File: python/720. Longest Word in Dictionary.py
class Solution:
    def longestWord(self, words: List[str]) -> str:
        words.sort()
        ws = set()
        is_cont = [False] * len(words)
        ans = ""

        for i, w in enumerate(words):
            if len(w) == 1:
                ws.add(w)
                if not ans:
                    ans = w
                ws.add(w)
            elif w[:-1] in ws:
                if len(w) > len(ans):
                    ans = w
                ws.add(w)
        return ans

File: rust/720. Longest Word in Dictionary.rs
impl Solution {
    pub fn longest_word(mut words: Vec<String>) -> String {
        words.sort();
        let mut ws = std::collections::HashSet::new();
        let mut ans = "".to_string();
        for w in &words {
            if w.len() == 1 {
                ws.insert(w);
                if ans.is_empty() {
                    ans = w.to_string();
                }
            } else {
                if ws.contains(&w[..w.len() - 1].to_string()) {
                    ws.insert(w);
                    if ans.len() < w.len() {
                        ans = w.to_string();
                    }
                }
            }
        }
        ans
    }
}
